oximeter/instruments/src/kstat/mod.rs

```diff
@@ -262,7 +275,35 @@ impl<'a> ConvertNamedData for NamedData<'a> {
     }
 }
 
-#[link(name = "c")]
-extern "C" {
-    fn gethrtime() -> i64;
+#[cfg(target_os = "illumos")]
+mod time {
+    #[link(name = "c")]
+    extern "C" {
+        fn gethrtime() -> i64;
```

Comment:
No objection here but just curious about the difference - is there a preference for this over using `clock_gettime()`? (I guess the constant is different for illumos vs linux, which is a little annoying, but it'd be _less_ of a difference?) Or, is there a reason we can't use `Instant` and forego the platform-specific bits altogether?

Reply:
Yeah, I considered making the `cfg`-detection be around the clock type. I don't have a strong preference, so that might be more reasonable. `man clock_gettime` is clear that the high-res clock serves as the source for `gethrtime()` anyway, so it would be clearer and simpler to do it that way. Thanks.

Reply:
Reworked this in dca0355f3 so only the clock-source is platform-dependent